src/Chapter2.hs

```diff
@@ -858,7 +888,11 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind [] = []
+rewind [x] = [x]
+rewind (x:xs) = (last xs) : rewind([x] ++ take(length(xs) - 1) xs) 
```

Comment:
Your solution is correct! However, it is slow. In lists, it is quite slow to add anything at the end of the list or take the list element of the list. That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

Reply:
I understand, I was rotating in place which is very well  explained by the example of the train, when thinking about the problem I could not see how to solve it in another way, so I "brute force it", but now when looking at the other tasks I think I got it, I try to create a new array from reading the initial one